var/spack/repos/builtin/packages/openmpi/package.py

```diff
@@ -268,14 +270,14 @@ class Openmpi(AutotoolsPackage):
 
     depends_on('libevent@2.0:', when='@4:')
 
-    depends_on('hwloc@2.0:', when='@4:')
+    depends_on('hwloc@2.0:', when='@4: ~internal-hwloc')
     # ompi@:3.0.0 doesn't support newer hwloc releases:
     # "configure: error: OMPI does not currently support hwloc v2 API"
     # Future ompi releases may support it, needs to be verified.
     # See #7483 for context.
-    depends_on('hwloc@:1.999', when='@:3.999.9999')
+    depends_on('hwloc@:1.999', when='@:3.999.999 ~internal-hwloc')
 
-    depends_on('hwloc +cuda', when='+cuda')
+    depends_on('hwloc +cuda', when='cuda ~internal-hwloc')
```

Comment:
I think #21811 may help for the `clingo` error. My bet is that the error is an obscure way to say that here you need:
```suggestion
    depends_on('hwloc +cuda', when='+cuda ~internal-hwloc')
```